subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/transport/RepositoryTransportFactory.java

```diff
@@ -108,7 +113,7 @@ public RepositoryTransport createTransport(Set<String> schemes, String name, Col
         ExternalResourceConnector resourceConnector = connectorFactory.createResourceConnector(connectionDetails);
         ExternalResourceCachePolicy cachePolicy = startParameterResolutionOverride.overrideExternalResourceCachePolicy(new DefaultExternalResourceCachePolicy());
 
-        return new ResourceConnectorRepositoryTransport(name, progressLoggerFactory, temporaryFileProvider, cachedExternalResourceIndex, timeProvider, cacheLockingManager, resourceConnector, buildOperationExecutor, cachePolicy);
+        return new ResourceConnectorRepositoryTransport(name, progressLoggerFactory, temporaryFileProvider, cachedExternalResourceIndex, timeProvider, cacheLockingManager, resourceConnector, buildOperationExecutor, cachePolicy, producerGuard);
```

Comment:
The number of parameters is too damn high ;) I really wonder whether there should be some higher level services here that encapsulate some of the interactions between these lower-level parameters.